npm/qsharp/src/compiler/compiler.ts

```diff
@@ -277,6 +278,22 @@ export class Compiler implements ICompiler {
     return this.wasm.generate_docs();
   }
 
+  async getCombinedDocumentation(): Promise<string> {
+    const docFiles: IDocFile[] = this.wasm.generate_docs();
+    // Create combined documentation in the worker
+    let content = "";
+    for (const file of docFiles) {
+      // Some files may contain information other than documentation
+      // For example, table of content is a separate file in a special format
+      // We check presence of qsharp.name in metadata to make sure we take
+      // only files that contain documentation from some qsharp object.
+      if (file.metadata.indexOf("qsharp.name:") >= 0) {
+        content += file.contents + "\n---\n";
```

Comment:
It might be helpful to pass a collection of documents, rather than one giant concatenated string. I can imaging (especially once we have docs for user code or references packages) that we might want to sort, filter, collapse, etc. by namespace. If it's all one long string we just have to parse it and break it all apart again somehow.

Reply:
Updated. I wanted to put most processing in the worker. But I moved concatenation out of it now.